alacritty/src/window_context.rs

```diff
@@ -98,15 +99,15 @@ impl WindowContext {
         // This object contains all of the state about what's being displayed. It's
         // wrapped in a clonable mutex since both the I/O loop and display need to
         // access it.
-        let terminal = Term::new(&config.terminal_config, display.size_info, event_proxy.clone());
+        let terminal = Term::new(&config.terminal_config, &display.size_info, event_proxy.clone());
         let terminal = Arc::new(FairMutex::new(terminal));
 
         // Create the PTY.
         //
         // The PTY forks a process to run the shell on the slave side of the
         // pseudoterminal. A file descriptor for the master side is retained for
         // reading/writing to the shell.
-        let pty = tty::new(&pty_config, &display.size_info, display.window.x11_window_id())?;
+        let pty = tty::new(&pty_config, display.size_info.into(), display.window.x11_window_id())?;
```

Comment:
What's the point of having `WinSize` when all we do with it is use it to convert `SizeInfo` into it, which then is just used to create `libc`'s `winsize`? Might as well just take `winsize` or `impl ToWinsize` as parameter and save yourself the extra useless type.

Reply:
It's due to Windows, since it doesn't have libc::winsize.

Reply:
The trait could still be used, right?

Reply:
The thing is that this type should be somewhere, since windows should have some inner type for that. The trait is used under the hood for unix anyway.

How we should proceed with it?

Reply:
Can you not take a trait impl instead of a custom type plus a trait impl?